web/packages/teleterm/src/ui/ConnectMyComputer/connectMyComputerContext.tsx

```diff
@@ -277,6 +298,7 @@ export const ConnectMyComputerContextProvider: FC<{
         downloadAndStartAgent,
         markAgentAsConfigured,
         isAgentConfiguredAttempt,
+        logsFromStartTimeout,
```

Comment:
I wasn't able to avoid exporting this, solely due to the fact that in the setup tab we don't have access to `currentAction` and we have to operate entirely on an attempt alone.

https://github.com/gravitational/teleport/blob/01a6f92322a71ae69f4a2db4d4bd18915f950a69/web/packages/teleterm/src/ui/ConnectMyComputer/DocumentConnectMyComputerSetup/DocumentConnectMyComputerSetup.tsx#L237-L244

Long term if we run into a problem like this again, we should just expand `useAsync` so that other than `statusText`, it also includes some kind of an `error` field that holds the whole error. OTOH, an error is not serializable so idk if we can freely put it there.

But ultimately the problem is that `useAsync` is able to represent the error state merely with a string (`statusText`) and that is not enough in some cases. But maybe it's fine to just have "out of band" pieces of state in cases like that. We lose some benefits coming from types, but at least we don't have problems with serializing `useAttempt` attempt object.

Reply:
I came across this issue recently with the unified resources call. I wanted to know the specific error being returned from the backend because a 404 or 501 would mean "not implemented" which I could then push around to the legacy view, rather than some actual server issue. I ended up catching it after the call (and had access to the error there) but I agree that it would have been easier if we could just directly access it somewhere in `useAsync` similar to how things like `react-query` work.

Reply:
AFAIK the state doesn't have to be serializable.

+1 for adding `error` to `useAsync`. I remember that I also had a problem with lack of it.